(winscp.install) Upgrade on Win11 yields `Cannot convert value to type System.String`

### Chocolatey Version

2.2.2

### Chocolatey License

None

### Package Version

Upgrade from 6.1.2 to 6.3.0

### Current Behaviour

Upgrading on a Windows 11 machine yields a `ERROR: Cannot process argument transformation on parameter 'file64'. Cannot convert value to type System.String.`

Upgrading some Windows 10 instances worked properly.

### Expected Behaviour

_No response_

### Steps To Reproduce

run a `choco upgrade winscp`
For the log below I ran a `choco upgrade winscp.install` as upgrade of the winscp metapackage succeded despite winscp.install wasn't upgraded.

### Environment

```markdown
- Operating System: Microsoft Windows NT 10.0.22631.0
- PowerShell Version: 5.1.22621.2506
- Shell: Windows PowerShell
```


### Chocolatey Log

```shell
2024-02-20 13:41:24,363 13516 [INFO ] - ============================================================
2024-02-20 13:41:24,702 13516 [INFO ] - Chocolatey v2.2.2
2024-02-20 13:41:24,709 13516 [DEBUG] - Chocolatey is running on Windows v 10.0.22631.0
2024-02-20 13:41:24,711 13516 [DEBUG] - Attempting to delete file "C:/ProgramData/chocolatey/choco.exe.old".
2024-02-20 13:41:24,712 13516 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\choco.exe.old".
2024-02-20 13:41:24,720 13516 [DEBUG] - Command line: "C:\ProgramData\chocolatey\choco.exe" upgrade winscp.install
2024-02-20 13:41:24,721 13516 [DEBUG] - Received arguments: upgrade winscp.install
2024-02-20 13:41:24,754 13516 [DEBUG] - RemovePendingPackagesTask is now ready and waiting for PreRunMessage.
2024-02-20 13:41:24,761 13516 [DEBUG] - Sending message 'PreRunMessage' out if there are subscribers...
2024-02-20 13:41:24,767 13516 [DEBUG] - [Pending] Removing all pending packages that should not be considered installed...
2024-02-20 13:41:24,797 13516 [DEBUG] - Performing validation checks.
2024-02-20 13:41:24,801 13516 [DEBUG] - Global Configuration Validation Checks:
2024-02-20 13:41:24,802 13516 [DEBUG] -  - Package Exit Code / Exit On Reboot = Checked
2024-02-20 13:41:24,805 13516 [DEBUG] - System State Validation Checks:
2024-02-20 13:41:24,807 13516 [DEBUG] -  Reboot Requirement Checks:
2024-02-20 13:41:24,809 13516 [DEBUG] -  - Pending Computer Rename = Checked
2024-02-20 13:41:24,811 13516 [DEBUG] -  - Pending Component Based Servicing = Checked
2024-02-20 13:41:24,812 13516 [DEBUG] -  - Pending Windows Auto Update = Checked
2024-02-20 13:41:24,814 13516 [DEBUG] -  - Pending File Rename Operations = Ignored
2024-02-20 13:41:24,816 13516 [DEBUG] -  - Pending Windows Package Installer = Checked
2024-02-20 13:41:24,817 13516 [DEBUG] -  - Pending Windows Package Installer SysWow64 = Checked
2024-02-20 13:41:24,821 13516 [DEBUG] - Cache Folder Lockdown Checks:
2024-02-20 13:41:24,822 13516 [DEBUG] -  - Elevated State = Checked
2024-02-20 13:41:24,822 13516 [DEBUG] -  - Folder Exists = Checked
2024-02-20 13:41:24,826 13516 [DEBUG] -  - Folder lockdown = Checked
2024-02-20 13:41:24,829 13516 [INFO ] - 3 validations performed. 3 success(es), 0 warning(s), and 0 error(s).
2024-02-20 13:41:24,839 13516 [DEBUG] - The source 'https://community.chocolatey.org/api/v2/' evaluated to a 'normal' source type
2024-02-20 13:41:24,841 13516 [DEBUG] - 
NOTE: Hiding sensitive configuration data! Please double and triple
 check to be sure no sensitive data is shown, especially if copying
 output to a gist for review.
2024-02-20 13:41:24,848 13516 [DEBUG] - Configuration: CommandName='upgrade'|
CacheLocation='C:\Users\[...]\AppData\Local\Temp\chocolatey'|
CommandExecutionTimeoutSeconds='2700'|WebRequestTimeoutSeconds='30'|
Sources='https://community.chocolatey.org/api/v2/'|SourceType='normal'|
ShowOnlineHelp='False'|Debug='False'|Verbose='False'|Trace='False'|
Force='False'|Noop='False'|HelpRequested='False'|
UnsuccessfulParsing='False'|RegularOutput='True'|QuietOutput='False'|
PromptForConfirmation='False'|DisableCompatibilityChecks='False'|
AcceptLicense='False'|AllowUnofficialBuild='False'|
Input='winscp.install'|AllVersions='False'|
SkipPackageInstallProvider='False'|SkipHookScripts='False'|
PackageNames='winscp.install'|Prerelease='False'|ForceX86='False'|
OverrideArguments='False'|NotSilent='False'|
ApplyPackageParametersToDependencies='False'|
ApplyInstallArgumentsToDependencies='False'|IgnoreDependencies='False'|
CacheExpirationInMinutes='30'|AllowDowngrade='False'|
ForceDependencies='False'|PinPackage='False'|
Information.PlatformType='Windows'|
Information.PlatformVersion='10.0.22631.0'|
Information.PlatformName='Windows 10'|
Information.ChocolateyVersion='2.2.2.0'|
Information.ChocolateyProductVersion='2.2.2'|
Information.FullName='choco, Version=2.2.2.0, Culture=neutral, PublicKeyToken=[...]'|
Information.Is64BitOperatingSystem='True'|
Information.Is64BitProcess='True'|Information.IsInteractive='True'|
Information.IsUserAdministrator='True'|
Information.IsUserSystemAccount='False'|
Information.IsUserRemoteDesktop='False'|
Information.IsUserRemote='False'|Information.IsProcessElevated='True'|
Information.IsLicensedVersion='False'|
Information.IsLicensedAssemblyLoaded='False'|
Information.LicenseType='Foss'|
Information.CurrentDirectory='C:\Users\[...]'|
Features.AutoUninstaller='True'|Features.ChecksumFiles='True'|
Features.AllowEmptyChecksums='False'|
Features.AllowEmptyChecksumsSecure='True'|
Features.FailOnAutoUninstaller='False'|
Features.FailOnStandardError='False'|Features.UsePowerShellHost='True'|
Features.LogEnvironmentValues='False'|Features.LogWithoutColor='False'|
Features.VirusCheck='False'|
Features.FailOnInvalidOrMissingLicense='False'|
Features.IgnoreInvalidOptionsSwitches='True'|
Features.UsePackageExitCodes='True'|
Features.UseEnhancedExitCodes='False'|
Features.UseFipsCompliantChecksums='False'|
Features.ShowNonElevatedWarnings='True'|
Features.ShowDownloadProgress='True'|
Features.StopOnFirstPackageFailure='False'|
Features.UseRememberedArgumentsForUpgrades='False'|
Features.IgnoreUnfoundPackagesOnUpgradeOutdated='False'|
Features.SkipPackageUpgradesWhenNotInstalled='False'|
Features.RemovePackageInformationOnUninstall='False'|
Features.ExitOnRebootDetected='False'|
Features.LogValidationResultsOnWarnings='True'|
Features.UsePackageRepositoryOptimizations='True'|
ListCommand.LocalOnly='False'|ListCommand.IdOnly='False'|
ListCommand.IncludeRegistryPrograms='False'|ListCommand.PageSize='25'|
ListCommand.Exact='False'|ListCommand.ByIdOnly='False'|
ListCommand.ByTagOnly='False'|ListCommand.IdStartsWith='False'|
ListCommand.OrderByPopularity='False'|ListCommand.ApprovedOnly='False'|
ListCommand.DownloadCacheAvailable='False'|
ListCommand.NotBroken='False'|
ListCommand.IncludeVersionOverrides='False'|
ListCommand.ExplicitPageSize='False'|
ListCommand.ExplicitSource='False'|
UpgradeCommand.FailOnUnfound='False'|
UpgradeCommand.FailOnNotInstalled='False'|
UpgradeCommand.NotifyOnlyAvailableUpgrades='False'|
UpgradeCommand.ExcludePrerelease='False'|
NewCommand.AutomaticPackage='False'|
NewCommand.UseOriginalTemplate='False'|SourceCommand.Command='unknown'|
SourceCommand.Priority='0'|SourceCommand.BypassProxy='False'|
SourceCommand.AllowSelfService='False'|
SourceCommand.VisibleToAdminsOnly='False'|
FeatureCommand.Command='unknown'|ConfigCommand.Command='Unknown'|
ApiKeyCommand.Command='Unknown'|PinCommand.Command='Unknown'|
OutdatedCommand.IgnorePinned='False'|
ExportCommand.IncludeVersionNumbers='False'|Proxy.BypassOnLocal='True'|
TemplateCommand.Command='unknown'|CacheCommand.Command='Unknown'|
CacheCommand.RemoveExpiredItemsOnly='False'|
2024-02-20 13:41:24,849 13516 [DEBUG] - _ Chocolatey:ChocolateyUpgradeCommand - Normal Run Mode _
2024-02-20 13:41:24,854 13516 [INFO ] - Upgrading the following packages:
2024-02-20 13:41:24,856 13516 [INFO ] - winscp.install
2024-02-20 13:41:24,857 13516 [INFO ] - By upgrading, you accept licenses for the packages.
2024-02-20 13:41:24,985 13516 [DEBUG] - Running list with the following filter = ''
2024-02-20 13:41:24,986 13516 [DEBUG] - --- Start of List ---
2024-02-20 13:41:25,002 13516 [DEBUG] - Resolving resource PackageSearchResource for source C:\ProgramData\chocolatey\lib
2024-02-20 13:41:17,016 17484 [DEBUG] - chocolatey-core.extension 1.4.0
[..]
2024-02-20 13:41:25,427 13516 [DEBUG] - winscp 6.3.0
2024-02-20 13:41:25,428 13516 [DEBUG] - winscp.install 6.1.2
2024-02-20 13:41:25,434 13516 [DEBUG] - --- End of List ---
2024-02-20 13:41:25,462 13516 [DEBUG] - Resolving resource ListResource for source https://community.chocolatey.org/api/v2/
2024-02-20 13:41:25,717 13516 [INFO ] - [NuGet]   CACHE https://community.chocolatey.org/api/v2/$metadata
2024-02-20 13:41:25,737 13516 [INFO ] - [NuGet]   CACHE https://community.chocolatey.org/api/v2/Packages()?$filter=(tolower(Id)%20eq%20'winscp.install')%20and%20IsLatestVersion&semVerLevel=2.0.0
2024-02-20 13:41:25,754 13516 [WARN ] - 
You have winscp.install v6.1.2 installed. Version 6.3.0 is available based on your source(s).
2024-02-20 13:41:25,759 13516 [DEBUG] - Resolving resource DependencyInfoResource for source https://community.chocolatey.org/api/v2/
2024-02-20 13:41:25,768 13516 [INFO ] - [NuGet]   CACHE https://community.chocolatey.org/api/v2/Packages(Id='winscp.install',Version='6.3.0')
2024-02-20 13:41:25,786 13516 [INFO ] - [NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='chocolatey-core.extension'&semVerLevel=2.0.0
2024-02-20 13:41:25,791 13516 [INFO ] - [NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='chocolatey-compatibility.extension'&semVerLevel=2.0.0
2024-02-20 13:41:25,796 13516 [INFO ] - [NuGet]   CACHE https://community.chocolatey.org/api/v2/Packages()?$filter=(tolower(Id)%20eq%20'winscp')%20and%20IsLatestVersion&semVerLevel=2.0.0
2024-02-20 13:41:25,799 13516 [INFO ] - [NuGet]   CACHE https://community.chocolatey.org/api/v2/Packages(Id='winscp',Version='6.3.0')
2024-02-20 13:41:25,801 13516 [INFO ] - [NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='winscp.install'&semVerLevel=2.0.0
2024-02-20 13:41:25,807 13516 [INFO ] - [NuGet]   CACHE http://community.chocolatey.org/api/v2/FindPackagesById?id='winscp.install'&$skiptoken='winscp.install','5.21.2'
2024-02-20 13:41:25,813 13516 [INFO ] - [NuGet]   CACHE https://community.chocolatey.org/api/v2/FindPackagesById()?id='chocolatey-core.extension'&semVerLevel=2.0.0
2024-02-20 13:41:25,918 13516 [WARN ] - [NuGet] One or more unresolved package dependency constraints detected in the Chocolatey lib folder. All dependency constraints must be resolved to add or update packages. If these packages are being updated this message may be ignored, if not the following error(s) may be blocking the current package operation: 'winscp 6.3.0 constraint: winscp.install (= 6.3.0)'
2024-02-20 13:41:25,975 13516 [INFO ] - [NuGet] Resolving dependency information took 0 ms
2024-02-20 13:41:26,138 13516 [DEBUG] - Running beforeModify step for 'winscp.install'
2024-02-20 13:41:26,145 13516 [DEBUG] - Backing up package files for 'winscp.install'
2024-02-20 13:41:26,198 13516 [DEBUG] - Attempting to create directory "C:\ProgramData\chocolatey\lib-bkp\winscp.install".
2024-02-20 13:41:26,200 13516 [DEBUG] - Moving C:\ProgramData\chocolatey\lib\winscp.install to C:\ProgramData\chocolatey\lib-bkp\winscp.install\6.1.2
2024-02-20 13:41:26,202 13516 [DEBUG] - Moving 'C:\ProgramData\chocolatey\lib\winscp.install'
 to 'C:\ProgramData\chocolatey\lib-bkp\winscp.install\6.1.2'
2024-02-20 13:41:28,209 13516 [DEBUG] - Attempting to create directory "C:\ProgramData\chocolatey\lib\winscp.install".
2024-02-20 13:41:28,218 13516 [DEBUG] - Attempting to copy "C:\ProgramData\chocolatey\lib-bkp\winscp.install\6.1.2\.chocolateyPending"
 to "C:\ProgramData\chocolatey\lib\winscp.install\.chocolateyPending".
2024-02-20 13:41:28,227 13516 [DEBUG] - Attempting to copy "C:\ProgramData\chocolatey\lib-bkp\winscp.install\6.1.2\winscp.install.nupkg"
 to "C:\ProgramData\chocolatey\lib\winscp.install\winscp.install.nupkg".
2024-02-20 13:41:28,245 13516 [DEBUG] - Attempting to copy "C:\ProgramData\chocolatey\lib-bkp\winscp.install\6.1.2\winscp.install.nuspec"
 to "C:\ProgramData\chocolatey\lib\winscp.install\winscp.install.nuspec".
2024-02-20 13:41:28,251 13516 [DEBUG] - Attempting to create directory "C:\ProgramData\chocolatey\lib\winscp.install\legal".
2024-02-20 13:41:28,255 13516 [DEBUG] - Attempting to copy "C:\ProgramData\chocolatey\lib-bkp\winscp.install\6.1.2\legal\license.txt"
 to "C:\ProgramData\chocolatey\lib\winscp.install\legal\license.txt".
2024-02-20 13:41:28,259 13516 [DEBUG] - Attempting to copy "C:\ProgramData\chocolatey\lib-bkp\winscp.install\6.1.2\legal\VERIFICATION.txt"
 to "C:\ProgramData\chocolatey\lib\winscp.install\legal\VERIFICATION.txt".
2024-02-20 13:41:28,264 13516 [DEBUG] - Attempting to create directory "C:\ProgramData\chocolatey\lib\winscp.install\tools".
2024-02-20 13:41:28,268 13516 [DEBUG] - Attempting to copy "C:\ProgramData\chocolatey\lib-bkp\winscp.install\6.1.2\tools\chocolateyInstall.ps1"
 to "C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyInstall.ps1".
2024-02-20 13:41:28,271 13516 [DEBUG] - Attempting to copy "C:\ProgramData\chocolatey\lib-bkp\winscp.install\6.1.2\tools\chocolateyUninstall.ps1"
 to "C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyUninstall.ps1".
2024-02-20 13:41:28,274 13516 [DEBUG] - Attempting to copy "C:\ProgramData\chocolatey\lib-bkp\winscp.install\6.1.2\tools\WinSCP-6.1.2-Setup.exe"
 to "C:\ProgramData\chocolatey\lib\winscp.install\tools\WinSCP-6.1.2-Setup.exe".
2024-02-20 13:41:28,291 13516 [DEBUG] - Attempting to copy "C:\ProgramData\chocolatey\lib-bkp\winscp.install\6.1.2\tools\WinSCP-6.1.2-Setup.exe.ignore"
 to "C:\ProgramData\chocolatey\lib\winscp.install\tools\WinSCP-6.1.2-Setup.exe.ignore".
2024-02-20 13:41:29,800 13516 [DEBUG] - Deleting file C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyInstall.ps1
2024-02-20 13:41:29,803 13516 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyInstall.ps1".
2024-02-20 13:41:29,807 13516 [DEBUG] - Deleting file C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyUninstall.ps1
2024-02-20 13:41:29,810 13516 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyUninstall.ps1".
2024-02-20 13:41:29,820 13516 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\lib\winscp.install\tools\WinSCP-6.1.2-Setup.exe.ignore".
2024-02-20 13:41:29,825 13516 [DEBUG] - Ensuring removal of installation files.
2024-02-20 13:41:29,828 13516 [DEBUG] - Removing nupkg if it still exists.
2024-02-20 13:41:29,830 13516 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\lib\winscp.install\winscp.install.nupkg".
2024-02-20 13:41:29,833 13516 [DEBUG] - Resolving resource DownloadResource for source https://community.chocolatey.org/api/v2/
2024-02-20 13:41:29,870 13516 [DEBUG] - Attempting to delete file "".
2024-02-20 13:41:29,881 13516 [INFO ] - [NuGet]   GET https://community.chocolatey.org/api/v2/package/winscp.install/6.3.0
2024-02-20 13:41:30,998 13516 [INFO ] - [NuGet]   OK https://community.chocolatey.org/api/v2/package/winscp.install/6.3.0 1115ms
2024-02-20 13:41:31,022 13516 [INFO ] - [NuGet] Acquiring lock for the installation of winscp.install 6.3.0
2024-02-20 13:41:31,045 13516 [INFO ] - [NuGet] Acquired lock for the installation of winscp.install 6.3.0
2024-02-20 13:41:33,498 13516 [INFO ] - [NuGet] Installed winscp.install 6.3.0 from https://community.chocolatey.org/api/v2/ with content hash uUZ3I+VjnfII4KqAon9hO0q/ln5RPIQwBOTnCRKPebnuQQAFJtBjGV8EaV9TaXlWgI41kLR7vA5ugWs3Cy5Xrw==.
2024-02-20 13:41:33,580 13516 [INFO ] - [NuGet] Adding package 'winscp.install.6.3.0 : chocolatey-core.extension [1.3.3, )' to folder 'C:\ProgramData\chocolatey\lib'
2024-02-20 13:41:34,711 13516 [INFO ] - [NuGet] Added package 'winscp.install.6.3.0 : chocolatey-core.extension [1.3.3, )' to folder 'C:\ProgramData\chocolatey\lib'
2024-02-20 13:41:34,716 13516 [DEBUG] - Attempting to delete file "C:\Users\[...]\AppData\Local\Temp\chocolatey\ChocolateyScratch\winscp.install/6.3.0\winscp.install.6.3.0.nupkg".
2024-02-20 13:41:34,719 13516 [DEBUG] - Attempting to delete file "C:\Users\[...]\AppData\Local\Temp\chocolatey\ChocolateyScratch\winscp.install/6.3.0\.nupkg.metadata".
2024-02-20 13:41:34,720 13516 [DEBUG] - Attempting to delete file "C:\Users\[...]\AppData\Local\Temp\chocolatey\ChocolateyScratch\winscp.install/6.3.0\winscp.install.6.3.0.nupkg.sha512".
2024-02-20 13:41:34,722 13516 [INFO ] - 
winscp.install v6.3.0 [Approved]
2024-02-20 13:41:34,746 13516 [INFO ] - winscp.install package files upgrade completed. Performing other installation steps.
2024-02-20 13:41:34,920 13516 [DEBUG] - Setting installer args for winscp.install
2024-02-20 13:41:34,921 13516 [DEBUG] - Setting package parameters for winscp.install
2024-02-20 13:41:34,922 13516 [DEBUG] - Contents of 'C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyInstall.ps1':
2024-02-20 13:41:34,929 13516 [DEBUG] - $ErrorActionPreference = 'Stop'

$toolsPath = Split-Path $MyInvocation.MyCommand.Definition

$packageArgs = @{
  packageName    = 'winscp'
  fileType       = 'exe'
  file           = Get-Item $toolsPath\*.exe
  file64         = Get-Item $toolsPath\*.exe
  silentArgs     = '/VERYSILENT /SUPPRESSMSGBOXES /NORESTART /SP-'
}
Install-ChocolateyInstallPackage @packageArgs
Get-ChildItem $toolsPath\*.exe | ForEach-Object { Remove-Item $_ -ea 0; if (Test-Path $_) { Set-Content "$_.ignore" '' }

$packageName = $packageArgs.packageName
$installLocation = Get-AppInstallLocation "$packageName*"
if (!$installLocation)  { Write-Warning "Can't find $packageName install location"; return }
Write-Host "$packageName installed to '$installLocation'"

Install-BinFile $packageName "$installLocation\$packageName.exe"
Install-BinFile "${packageName}.com" "${installLocation}\${packageName}.com"

Register-Application "$installLocation\$packageName.exe"
Write-Host "$packageName registered as $packageName"

2024-02-20 13:41:34,944 13516 [DEBUG] - Calling built-in PowerShell host with ['[System.Threading.Thread]::CurrentThread.CurrentCulture = '';[System.Threading.Thread]::CurrentThread.CurrentUICulture = '';[System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::SystemDefault; & import-module -name 'C:\ProgramData\chocolatey\helpers\chocolateyInstaller.psm1'; & 'C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1' -packageScript 'C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyInstall.ps1' -installArguments '' -packageParameters '' -preRunHookScripts $null -postRunHookScripts $null']
2024-02-20 13:41:34,954 13516 [DEBUG] - Redirecting System.Management.Automation.resources, Version=3.0.0.0, Culture=en-US, PublicKeyToken=[...], requested by ''
2024-02-20 13:41:36,136 13516 [DEBUG] - Host version is 5.1.22621.1, PowerShell Version is '5.1.22621.2506' and CLR Version is '4.0.30319.42000'.
2024-02-20 13:41:36,513 13516 [INFO ] - VERBOSE: Exporting function 'Format-FileSize'.
2024-02-20 13:41:36,514 13516 [INFO ] - VERBOSE: Exporting function 'Get-ChecksumValid'.
2024-02-20 13:41:36,516 13516 [INFO ] - VERBOSE: Exporting function 'Get-ChocolateyConfigValue'.
2024-02-20 13:41:36,517 13516 [INFO ] - VERBOSE: Exporting function 'Get-ChocolateyPath'.
2024-02-20 13:41:36,518 13516 [INFO ] - VERBOSE: Exporting function 'Get-ChocolateyUnzip'.
2024-02-20 13:41:36,518 13516 [INFO ] - VERBOSE: Exporting function 'Get-ChocolateyWebFile'.
2024-02-20 13:41:36,520 13516 [INFO ] - VERBOSE: Exporting function 'Get-EnvironmentVariable'.
2024-02-20 13:41:36,521 13516 [INFO ] - VERBOSE: Exporting function 'Get-EnvironmentVariableNames'.
2024-02-20 13:41:36,521 13516 [INFO ] - VERBOSE: Exporting function 'Get-FtpFile'.
2024-02-20 13:41:36,523 13516 [INFO ] - VERBOSE: Exporting function 'Get-OSArchitectureWidth'.
2024-02-20 13:41:36,524 13516 [INFO ] - VERBOSE: Exporting function 'Get-PackageParameters'.
2024-02-20 13:41:36,524 13516 [INFO ] - VERBOSE: Exporting function 'Get-PackageParametersBuiltIn'.
2024-02-20 13:41:36,525 13516 [INFO ] - VERBOSE: Exporting function 'Get-ToolsLocation'.
2024-02-20 13:41:36,526 13516 [INFO ] - VERBOSE: Exporting function 'Get-UACEnabled'.
2024-02-20 13:41:36,527 13516 [INFO ] - VERBOSE: Exporting function 'Get-UninstallRegistryKey'.
2024-02-20 13:41:36,528 13516 [INFO ] - VERBOSE: Exporting function 'Get-VirusCheckValid'.
2024-02-20 13:41:36,528 13516 [INFO ] - VERBOSE: Exporting function 'Get-WebFile'.
2024-02-20 13:41:36,529 13516 [INFO ] - VERBOSE: Exporting function 'Get-WebFileName'.
2024-02-20 13:41:36,530 13516 [INFO ] - VERBOSE: Exporting function 'Get-WebHeaders'.
2024-02-20 13:41:36,531 13516 [INFO ] - VERBOSE: Exporting function 'Install-BinFile'.
2024-02-20 13:41:36,532 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyEnvironmentVariable'.
2024-02-20 13:41:36,532 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyExplorerMenuItem'.
2024-02-20 13:41:36,534 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyFileAssociation'.
2024-02-20 13:41:36,535 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyInstallPackage'.
2024-02-20 13:41:36,535 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyPackage'.
2024-02-20 13:41:36,536 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyPath'.
2024-02-20 13:41:36,537 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyPinnedTaskBarItem'.
2024-02-20 13:41:36,538 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyPowershellCommand'.
2024-02-20 13:41:36,539 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyShortcut'.
2024-02-20 13:41:36,540 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyVsixPackage'.
2024-02-20 13:41:36,540 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyZipPackage'.
2024-02-20 13:41:36,541 13516 [INFO ] - VERBOSE: Exporting function 'Install-Vsix'.
2024-02-20 13:41:36,542 13516 [INFO ] - VERBOSE: Exporting function 'Set-EnvironmentVariable'.
2024-02-20 13:41:36,543 13516 [INFO ] - VERBOSE: Exporting function 'Set-PowerShellExitCode'.
2024-02-20 13:41:36,543 13516 [INFO ] - VERBOSE: Exporting function 'Start-ChocolateyProcessAsAdmin'.
2024-02-20 13:41:36,543 13516 [INFO ] - VERBOSE: Exporting function 'Test-ProcessAdminRights'.
2024-02-20 13:41:36,545 13516 [INFO ] - VERBOSE: Exporting function 'Uninstall-BinFile'.
2024-02-20 13:41:36,545 13516 [INFO ] - VERBOSE: Exporting function 'Uninstall-ChocolateyEnvironmentVariable'.
2024-02-20 13:41:36,546 13516 [INFO ] - VERBOSE: Exporting function 'Uninstall-ChocolateyPackage'.
2024-02-20 13:41:36,547 13516 [INFO ] - VERBOSE: Exporting function 'Uninstall-ChocolateyZipPackage'.
2024-02-20 13:41:36,547 13516 [INFO ] - VERBOSE: Exporting function 'Update-SessionEnvironment'.
2024-02-20 13:41:36,548 13516 [INFO ] - VERBOSE: Exporting function 'Write-FunctionCallLogMessage'.
2024-02-20 13:41:36,552 13516 [INFO ] - VERBOSE: Exporting alias 'Get-ProcessorBits'.
2024-02-20 13:41:36,553 13516 [INFO ] - VERBOSE: Exporting alias 'Get-OSBitness'.
2024-02-20 13:41:36,554 13516 [INFO ] - VERBOSE: Exporting alias 'Get-InstallRegistryKey'.
2024-02-20 13:41:36,555 13516 [INFO ] - VERBOSE: Exporting alias 'Generate-BinFile'.
2024-02-20 13:41:36,555 13516 [INFO ] - VERBOSE: Exporting alias 'Add-BinFile'.
2024-02-20 13:41:36,556 13516 [INFO ] - VERBOSE: Exporting alias 'Start-ChocolateyProcess'.
2024-02-20 13:41:36,557 13516 [INFO ] - VERBOSE: Exporting alias 'Invoke-ChocolateyProcess'.
2024-02-20 13:41:36,558 13516 [INFO ] - VERBOSE: Exporting alias 'Remove-BinFile'.
2024-02-20 13:41:36,559 13516 [INFO ] - VERBOSE: Exporting alias 'refreshenv'.
2024-02-20 13:41:36,568 13516 [DEBUG] - Loading community extensions
2024-02-20 13:41:36,600 13516 [DEBUG] - Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-compatibility\chocolatey-compatibility.psm1'
2024-02-20 13:41:36,601 13516 [INFO ] - VERBOSE: Loading module from path 'C:\ProgramData\chocolatey\extensions\chocolatey-compatibility\chocolatey-compatibility.psm1'.
2024-02-20 13:41:36,647 13516 [DEBUG] - Function 'Get-PackageParameters' exists, ignoring export.
2024-02-20 13:41:36,649 13516 [DEBUG] - Function 'Get-UninstallRegistryKey' exists, ignoring export.
2024-02-20 13:41:36,650 13516 [DEBUG] - Exporting function 'Install-ChocolateyDesktopLink' for backwards compatibility
2024-02-20 13:41:36,652 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyDesktopLink'.
2024-02-20 13:41:36,655 13516 [DEBUG] - Exporting function 'Write-ChocolateyFailure' for backwards compatibility
2024-02-20 13:41:36,658 13516 [INFO ] - VERBOSE: Exporting function 'Write-ChocolateyFailure'.
2024-02-20 13:41:36,659 13516 [DEBUG] - Exporting function 'Write-ChocolateySuccess' for backwards compatibility
2024-02-20 13:41:36,663 13516 [INFO ] - VERBOSE: Exporting function 'Write-ChocolateySuccess'.
2024-02-20 13:41:36,664 13516 [DEBUG] - Exporting function 'Write-FileUpdateLog' for backwards compatibility
2024-02-20 13:41:36,666 13516 [INFO ] - VERBOSE: Exporting function 'Write-FileUpdateLog'.
2024-02-20 13:41:36,680 13516 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyDesktopLink'.
2024-02-20 13:41:36,681 13516 [INFO ] - VERBOSE: Importing function 'Write-ChocolateyFailure'.
2024-02-20 13:41:36,682 13516 [INFO ] - VERBOSE: Importing function 'Write-ChocolateySuccess'.
2024-02-20 13:41:36,683 13516 [INFO ] - VERBOSE: Importing function 'Write-FileUpdateLog'.
2024-02-20 13:41:36,689 13516 [DEBUG] - Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-core\chocolatey-core.psm1'
2024-02-20 13:41:36,690 13516 [INFO ] - VERBOSE: Loading module from path 'C:\ProgramData\chocolatey\extensions\chocolatey-core\chocolatey-core.psm1'.
2024-02-20 13:41:36,740 13516 [INFO ] - VERBOSE: Exporting function 'Get-AppInstallLocation'.
2024-02-20 13:41:36,741 13516 [INFO ] - VERBOSE: Exporting function 'Get-AvailableDriveLetter'.
2024-02-20 13:41:36,743 13516 [INFO ] - VERBOSE: Exporting function 'Get-EffectiveProxy'.
2024-02-20 13:41:36,744 13516 [INFO ] - VERBOSE: Exporting function 'Get-PackageCacheLocation'.
2024-02-20 13:41:36,745 13516 [INFO ] - VERBOSE: Exporting function 'Get-WebContent'.
2024-02-20 13:41:36,746 13516 [INFO ] - VERBOSE: Exporting function 'Register-Application'.
2024-02-20 13:41:36,747 13516 [INFO ] - VERBOSE: Exporting function 'Remove-Process'.
2024-02-20 13:41:36,748 13516 [INFO ] - VERBOSE: Importing function 'Get-AppInstallLocation'.
2024-02-20 13:41:36,749 13516 [INFO ] - VERBOSE: Importing function 'Get-AvailableDriveLetter'.
2024-02-20 13:41:36,750 13516 [INFO ] - VERBOSE: Importing function 'Get-EffectiveProxy'.
2024-02-20 13:41:36,750 13516 [INFO ] - VERBOSE: Importing function 'Get-PackageCacheLocation'.
2024-02-20 13:41:36,753 13516 [INFO ] - VERBOSE: Importing function 'Get-WebContent'.
2024-02-20 13:41:36,754 13516 [INFO ] - VERBOSE: Importing function 'Register-Application'.
2024-02-20 13:41:36,755 13516 [INFO ] - VERBOSE: Importing function 'Remove-Process'.
2024-02-20 13:41:36,756 13516 [DEBUG] - Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-dotnetfx\chocolatey-dotnetfx.psm1'
2024-02-20 13:41:36,757 13516 [INFO ] - VERBOSE: Loading module from path 'C:\ProgramData\chocolatey\extensions\chocolatey-dotnetfx\chocolatey-dotnetfx.psm1'.
2024-02-20 13:41:36,792 13516 [INFO ] - VERBOSE: Exporting function 'Install-DotNetFramework'.
2024-02-20 13:41:36,793 13516 [INFO ] - VERBOSE: Exporting function 'Install-DotNetDevPack'.
2024-02-20 13:41:36,793 13516 [INFO ] - VERBOSE: Importing function 'Install-DotNetDevPack'.
2024-02-20 13:41:36,795 13516 [INFO ] - VERBOSE: Importing function 'Install-DotNetFramework'.
2024-02-20 13:41:36,796 13516 [DEBUG] - Importing 'C:\ProgramData\chocolatey\extensions\chocolatey-windowsupdate\chocolatey-windowsupdate.psm1'
2024-02-20 13:41:36,797 13516 [INFO ] - VERBOSE: Loading module from path 'C:\ProgramData\chocolatey\extensions\chocolatey-windowsupdate\chocolatey-windowsupdate.psm1'.
2024-02-20 13:41:36,848 13516 [INFO ] - VERBOSE: Exporting function 'Install-WindowsUpdate'.
2024-02-20 13:41:36,849 13516 [INFO ] - VERBOSE: Exporting function 'Test-WindowsUpdate'.
2024-02-20 13:41:36,850 13516 [INFO ] - VERBOSE: Importing function 'Install-WindowsUpdate'.
2024-02-20 13:41:36,851 13516 [INFO ] - VERBOSE: Importing function 'Test-WindowsUpdate'.
2024-02-20 13:41:36,852 13516 [INFO ] - VERBOSE: Exporting function 'Format-FileSize'.
2024-02-20 13:41:36,853 13516 [INFO ] - VERBOSE: Exporting function 'Get-ChecksumValid'.
2024-02-20 13:41:36,855 13516 [INFO ] - VERBOSE: Exporting function 'Get-ChocolateyConfigValue'.
2024-02-20 13:41:36,856 13516 [INFO ] - VERBOSE: Exporting function 'Get-ChocolateyPath'.
2024-02-20 13:41:36,857 13516 [INFO ] - VERBOSE: Exporting function 'Get-ChocolateyUnzip'.
2024-02-20 13:41:36,858 13516 [INFO ] - VERBOSE: Exporting function 'Get-ChocolateyWebFile'.
2024-02-20 13:41:36,858 13516 [INFO ] - VERBOSE: Exporting function 'Get-EnvironmentVariable'.
2024-02-20 13:41:36,859 13516 [INFO ] - VERBOSE: Exporting function 'Get-EnvironmentVariableNames'.
2024-02-20 13:41:36,860 13516 [INFO ] - VERBOSE: Exporting function 'Get-FtpFile'.
2024-02-20 13:41:36,861 13516 [INFO ] - VERBOSE: Exporting function 'Get-OSArchitectureWidth'.
2024-02-20 13:41:36,862 13516 [INFO ] - VERBOSE: Exporting function 'Get-PackageParameters'.
2024-02-20 13:41:36,864 13516 [INFO ] - VERBOSE: Exporting function 'Get-PackageParametersBuiltIn'.
2024-02-20 13:41:36,864 13516 [INFO ] - VERBOSE: Exporting function 'Get-ToolsLocation'.
2024-02-20 13:41:36,865 13516 [INFO ] - VERBOSE: Exporting function 'Get-UACEnabled'.
2024-02-20 13:41:36,866 13516 [INFO ] - VERBOSE: Exporting function 'Get-UninstallRegistryKey'.
2024-02-20 13:41:36,867 13516 [INFO ] - VERBOSE: Exporting function 'Get-VirusCheckValid'.
2024-02-20 13:41:36,868 13516 [INFO ] - VERBOSE: Exporting function 'Get-WebFile'.
2024-02-20 13:41:36,868 13516 [INFO ] - VERBOSE: Exporting function 'Get-WebFileName'.
2024-02-20 13:41:36,869 13516 [INFO ] - VERBOSE: Exporting function 'Get-WebHeaders'.
2024-02-20 13:41:36,870 13516 [INFO ] - VERBOSE: Exporting function 'Install-BinFile'.
2024-02-20 13:41:36,871 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyEnvironmentVariable'.
2024-02-20 13:41:36,871 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyExplorerMenuItem'.
2024-02-20 13:41:36,873 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyFileAssociation'.
2024-02-20 13:41:36,874 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyInstallPackage'.
2024-02-20 13:41:36,875 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyPackage'.
2024-02-20 13:41:36,876 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyPath'.
2024-02-20 13:41:36,877 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyPinnedTaskBarItem'.
2024-02-20 13:41:36,878 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyPowershellCommand'.
2024-02-20 13:41:36,879 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyShortcut'.
2024-02-20 13:41:36,880 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyVsixPackage'.
2024-02-20 13:41:36,881 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyZipPackage'.
2024-02-20 13:41:36,881 13516 [INFO ] - VERBOSE: Exporting function 'Install-Vsix'.
2024-02-20 13:41:36,883 13516 [INFO ] - VERBOSE: Exporting function 'Set-EnvironmentVariable'.
2024-02-20 13:41:36,884 13516 [INFO ] - VERBOSE: Exporting function 'Set-PowerShellExitCode'.
2024-02-20 13:41:36,885 13516 [INFO ] - VERBOSE: Exporting function 'Start-ChocolateyProcessAsAdmin'.
2024-02-20 13:41:36,886 13516 [INFO ] - VERBOSE: Exporting function 'Test-ProcessAdminRights'.
2024-02-20 13:41:36,887 13516 [INFO ] - VERBOSE: Exporting function 'Uninstall-BinFile'.
2024-02-20 13:41:36,887 13516 [INFO ] - VERBOSE: Exporting function 'Uninstall-ChocolateyEnvironmentVariable'.
2024-02-20 13:41:36,888 13516 [INFO ] - VERBOSE: Exporting function 'Uninstall-ChocolateyPackage'.
2024-02-20 13:41:36,891 13516 [INFO ] - VERBOSE: Exporting function 'Uninstall-ChocolateyZipPackage'.
2024-02-20 13:41:36,892 13516 [INFO ] - VERBOSE: Exporting function 'Update-SessionEnvironment'.
2024-02-20 13:41:36,893 13516 [INFO ] - VERBOSE: Exporting function 'Write-FunctionCallLogMessage'.
2024-02-20 13:41:36,894 13516 [INFO ] - VERBOSE: Exporting function 'Install-ChocolateyDesktopLink'.
2024-02-20 13:41:36,895 13516 [INFO ] - VERBOSE: Exporting function 'Write-ChocolateyFailure'.
2024-02-20 13:41:36,897 13516 [INFO ] - VERBOSE: Exporting function 'Write-ChocolateySuccess'.
2024-02-20 13:41:36,898 13516 [INFO ] - VERBOSE: Exporting function 'Write-FileUpdateLog'.
2024-02-20 13:41:36,899 13516 [INFO ] - VERBOSE: Exporting function 'Get-AppInstallLocation'.
2024-02-20 13:41:36,900 13516 [INFO ] - VERBOSE: Exporting function 'Get-AvailableDriveLetter'.
2024-02-20 13:41:36,901 13516 [INFO ] - VERBOSE: Exporting function 'Get-EffectiveProxy'.
2024-02-20 13:41:36,902 13516 [INFO ] - VERBOSE: Exporting function 'Get-PackageCacheLocation'.
2024-02-20 13:41:36,903 13516 [INFO ] - VERBOSE: Exporting function 'Get-WebContent'.
2024-02-20 13:41:36,904 13516 [INFO ] - VERBOSE: Exporting function 'Register-Application'.
2024-02-20 13:41:36,906 13516 [INFO ] - VERBOSE: Exporting function 'Remove-Process'.
2024-02-20 13:41:36,907 13516 [INFO ] - VERBOSE: Exporting function 'Install-DotNetDevPack'.
2024-02-20 13:41:36,908 13516 [INFO ] - VERBOSE: Exporting function 'Install-DotNetFramework'.
2024-02-20 13:41:36,910 13516 [INFO ] - VERBOSE: Exporting function 'Install-WindowsUpdate'.
2024-02-20 13:41:36,911 13516 [INFO ] - VERBOSE: Exporting function 'Test-WindowsUpdate'.
2024-02-20 13:41:36,912 13516 [INFO ] - VERBOSE: Exporting alias 'Get-ProcessorBits'.
2024-02-20 13:41:36,913 13516 [INFO ] - VERBOSE: Exporting alias 'Get-OSBitness'.
2024-02-20 13:41:36,914 13516 [INFO ] - VERBOSE: Exporting alias 'Get-InstallRegistryKey'.
2024-02-20 13:41:36,915 13516 [INFO ] - VERBOSE: Exporting alias 'Generate-BinFile'.
2024-02-20 13:41:36,917 13516 [INFO ] - VERBOSE: Exporting alias 'Add-BinFile'.
2024-02-20 13:41:36,918 13516 [INFO ] - VERBOSE: Exporting alias 'Start-ChocolateyProcess'.
2024-02-20 13:41:36,919 13516 [INFO ] - VERBOSE: Exporting alias 'Invoke-ChocolateyProcess'.
2024-02-20 13:41:36,919 13516 [INFO ] - VERBOSE: Exporting alias 'Remove-BinFile'.
2024-02-20 13:41:36,921 13516 [INFO ] - VERBOSE: Exporting alias 'refreshenv'.
2024-02-20 13:41:36,922 13516 [INFO ] - VERBOSE: Importing function 'Format-FileSize'.
2024-02-20 13:41:36,924 13516 [INFO ] - VERBOSE: Importing function 'Get-AppInstallLocation'.
2024-02-20 13:41:36,925 13516 [INFO ] - VERBOSE: Importing function 'Get-AvailableDriveLetter'.
2024-02-20 13:41:36,927 13516 [INFO ] - VERBOSE: Importing function 'Get-ChecksumValid'.
2024-02-20 13:41:36,929 13516 [INFO ] - VERBOSE: Importing function 'Get-ChocolateyConfigValue'.
2024-02-20 13:41:36,930 13516 [INFO ] - VERBOSE: Importing function 'Get-ChocolateyPath'.
2024-02-20 13:41:36,932 13516 [INFO ] - VERBOSE: Importing function 'Get-ChocolateyUnzip'.
2024-02-20 13:41:36,933 13516 [INFO ] - VERBOSE: Importing function 'Get-ChocolateyWebFile'.
2024-02-20 13:41:36,935 13516 [INFO ] - VERBOSE: Importing function 'Get-EffectiveProxy'.
2024-02-20 13:41:36,936 13516 [INFO ] - VERBOSE: Importing function 'Get-EnvironmentVariable'.
2024-02-20 13:41:36,938 13516 [INFO ] - VERBOSE: Importing function 'Get-EnvironmentVariableNames'.
2024-02-20 13:41:36,939 13516 [INFO ] - VERBOSE: Importing function 'Get-FtpFile'.
2024-02-20 13:41:36,941 13516 [INFO ] - VERBOSE: Importing function 'Get-OSArchitectureWidth'.
2024-02-20 13:41:36,941 13516 [INFO ] - VERBOSE: Importing function 'Get-PackageCacheLocation'.
2024-02-20 13:41:36,943 13516 [INFO ] - VERBOSE: Importing function 'Get-PackageParameters'.
2024-02-20 13:41:36,945 13516 [INFO ] - VERBOSE: Importing function 'Get-PackageParametersBuiltIn'.
2024-02-20 13:41:36,946 13516 [INFO ] - VERBOSE: Importing function 'Get-ToolsLocation'.
2024-02-20 13:41:36,948 13516 [INFO ] - VERBOSE: Importing function 'Get-UACEnabled'.
2024-02-20 13:41:36,949 13516 [INFO ] - VERBOSE: Importing function 'Get-UninstallRegistryKey'.
2024-02-20 13:41:36,951 13516 [INFO ] - VERBOSE: Importing function 'Get-VirusCheckValid'.
2024-02-20 13:41:36,952 13516 [INFO ] - VERBOSE: Importing function 'Get-WebContent'.
2024-02-20 13:41:36,954 13516 [INFO ] - VERBOSE: Importing function 'Get-WebFile'.
2024-02-20 13:41:36,955 13516 [INFO ] - VERBOSE: Importing function 'Get-WebFileName'.
2024-02-20 13:41:36,956 13516 [INFO ] - VERBOSE: Importing function 'Get-WebHeaders'.
2024-02-20 13:41:36,957 13516 [INFO ] - VERBOSE: Importing function 'Install-BinFile'.
2024-02-20 13:41:36,958 13516 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyDesktopLink'.
2024-02-20 13:41:36,960 13516 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyEnvironmentVariable'.
2024-02-20 13:41:36,961 13516 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyExplorerMenuItem'.
2024-02-20 13:41:36,962 13516 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyFileAssociation'.
2024-02-20 13:41:36,963 13516 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyInstallPackage'.
2024-02-20 13:41:36,964 13516 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyPackage'.
2024-02-20 13:41:36,965 13516 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyPath'.
2024-02-20 13:41:36,966 13516 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyPinnedTaskBarItem'.
2024-02-20 13:41:36,967 13516 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyPowershellCommand'.
2024-02-20 13:41:36,968 13516 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyShortcut'.
2024-02-20 13:41:36,969 13516 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyVsixPackage'.
2024-02-20 13:41:36,970 13516 [INFO ] - VERBOSE: Importing function 'Install-ChocolateyZipPackage'.
2024-02-20 13:41:36,971 13516 [INFO ] - VERBOSE: Importing function 'Install-DotNetDevPack'.
2024-02-20 13:41:36,971 13516 [INFO ] - VERBOSE: Importing function 'Install-DotNetFramework'.
2024-02-20 13:41:36,972 13516 [INFO ] - VERBOSE: Importing function 'Install-Vsix'.
2024-02-20 13:41:36,973 13516 [INFO ] - VERBOSE: Importing function 'Install-WindowsUpdate'.
2024-02-20 13:41:36,974 13516 [INFO ] - VERBOSE: Importing function 'Register-Application'.
2024-02-20 13:41:36,975 13516 [INFO ] - VERBOSE: Importing function 'Remove-Process'.
2024-02-20 13:41:36,976 13516 [INFO ] - VERBOSE: Importing function 'Set-EnvironmentVariable'.
2024-02-20 13:41:36,977 13516 [INFO ] - VERBOSE: Importing function 'Set-PowerShellExitCode'.
2024-02-20 13:41:36,978 13516 [INFO ] - VERBOSE: Importing function 'Start-ChocolateyProcessAsAdmin'.
2024-02-20 13:41:36,979 13516 [INFO ] - VERBOSE: Importing function 'Test-ProcessAdminRights'.
2024-02-20 13:41:36,980 13516 [INFO ] - VERBOSE: Importing function 'Test-WindowsUpdate'.
2024-02-20 13:41:36,981 13516 [INFO ] - VERBOSE: Importing function 'Uninstall-BinFile'.
2024-02-20 13:41:36,981 13516 [INFO ] - VERBOSE: Importing function 'Uninstall-ChocolateyEnvironmentVariable'.
2024-02-20 13:41:36,982 13516 [INFO ] - VERBOSE: Importing function 'Uninstall-ChocolateyPackage'.
2024-02-20 13:41:36,982 13516 [INFO ] - VERBOSE: Importing function 'Uninstall-ChocolateyZipPackage'.
2024-02-20 13:41:36,983 13516 [INFO ] - VERBOSE: Importing function 'Update-SessionEnvironment'.
2024-02-20 13:41:36,984 13516 [INFO ] - VERBOSE: Importing function 'Write-ChocolateyFailure'.
2024-02-20 13:41:36,985 13516 [INFO ] - VERBOSE: Importing function 'Write-ChocolateySuccess'.
2024-02-20 13:41:36,986 13516 [INFO ] - VERBOSE: Importing function 'Write-FileUpdateLog'.
2024-02-20 13:41:36,987 13516 [INFO ] - VERBOSE: Importing function 'Write-FunctionCallLogMessage'.
2024-02-20 13:41:36,988 13516 [INFO ] - VERBOSE: Importing alias 'Add-BinFile'.
2024-02-20 13:41:36,989 13516 [INFO ] - VERBOSE: Importing alias 'Generate-BinFile'.
2024-02-20 13:41:36,990 13516 [INFO ] - VERBOSE: Importing alias 'Get-InstallRegistryKey'.
2024-02-20 13:41:36,991 13516 [INFO ] - VERBOSE: Importing alias 'Get-OSBitness'.
2024-02-20 13:41:36,992 13516 [INFO ] - VERBOSE: Importing alias 'Get-ProcessorBits'.
2024-02-20 13:41:36,993 13516 [INFO ] - VERBOSE: Importing alias 'Invoke-ChocolateyProcess'.
2024-02-20 13:41:36,994 13516 [INFO ] - VERBOSE: Importing alias 'refreshenv'.
2024-02-20 13:41:36,995 13516 [INFO ] - VERBOSE: Importing alias 'Remove-BinFile'.
2024-02-20 13:41:36,996 13516 [INFO ] - VERBOSE: Importing alias 'Start-ChocolateyProcess'.
2024-02-20 13:41:37,046 13516 [DEBUG] - ---------------------------Script Execution---------------------------
2024-02-20 13:41:37,048 13516 [DEBUG] - Running 'ChocolateyScriptRunner' for winscp.install v6.3.0 with packageScript 'C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyInstall.ps1', packageFolder:'C:\ProgramData\chocolatey\lib\winscp.install', installArguments: '', packageParameters: '', preRunHookScripts: '', postRunHookScripts: '',
2024-02-20 13:41:37,057 13516 [DEBUG] - Running package script 'C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyInstall.ps1'
2024-02-20 13:41:37,102 13516 [ERROR] - ERROR: Cannot process argument transformation on parameter 'file64'. Cannot convert value to type System.String.
2024-02-20 13:41:37,126 13516 [DEBUG] - Built-in PowerShell host called with ['[System.Threading.Thread]::CurrentThread.CurrentCulture = '';[System.Threading.Thread]::CurrentThread.CurrentUICulture = '';[System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::SystemDefault; & import-module -name 'C:\ProgramData\chocolatey\helpers\chocolateyInstaller.psm1'; & 'C:\ProgramData\chocolatey\helpers\chocolateyScriptRunner.ps1' -packageScript 'C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyInstall.ps1' -installArguments '' -packageParameters '' -preRunHookScripts $null -postRunHookScripts $null'] exited with '-1'.
2024-02-20 13:41:37,129 13516 [DEBUG] - Calling command ['"C:\WINDOWS\System32\shutdown.exe" /a']
2024-02-20 13:41:37,165 13516 [DEBUG] - Command ['"C:\WINDOWS\System32\shutdown.exe" /a'] exited with '1116'
2024-02-20 13:41:37,340 13516 [DEBUG] - Capturing package files in 'C:\ProgramData\chocolatey\lib\winscp.install'
2024-02-20 13:41:37,368 13516 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\winscp.install\winscp.install.nupkg'
  with checksum '0AE014168264DC06646F9E877733EA6B'
2024-02-20 13:41:37,370 13516 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\winscp.install\winscp.install.nuspec'
  with checksum '8A070150A866FC02EE4273E4991C7B9D'
2024-02-20 13:41:37,372 13516 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\winscp.install\legal\license.txt'
  with checksum '3C34AFDC3ADF82D2448F12715A255122'
2024-02-20 13:41:37,373 13516 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\winscp.install\legal\VERIFICATION.txt'
  with checksum 'D46A01AA2D836FBCB53DD14EEB4C8B5B'
2024-02-20 13:41:37,374 13516 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyInstall.ps1'
  with checksum '40B1712FEEF84338EE00BA7EAD7CA5D3'
2024-02-20 13:41:37,375 13516 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyUninstall.ps1'
  with checksum 'B0436E943518C06B85D2AD251B2F6208'
2024-02-20 13:41:37,405 13516 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\winscp.install\tools\WinSCP-6.1.2-Setup.exe'
  with checksum 'A655323232F2DEFB99C4E975AB1EDCEC'
2024-02-20 13:41:37,441 13516 [DEBUG] -  Found 'C:\ProgramData\chocolatey\lib\winscp.install\tools\WinSCP-6.3-Setup.exe'
  with checksum 'FEB4532B45421105558FF7EA32668F78'
2024-02-20 13:41:37,459 13516 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\winscp.install.6.3.0\.arguments".
2024-02-20 13:41:37,460 13516 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\winscp.install.6.3.0\.extra".
2024-02-20 13:41:37,461 13516 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\winscp.install.6.3.0\.version".
2024-02-20 13:41:37,462 13516 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\winscp.install.6.3.0\.sxs".
2024-02-20 13:41:37,463 13516 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\.chocolatey\winscp.install.6.3.0\.pin".
2024-02-20 13:41:37,466 13516 [DEBUG] - Attempting to delete directory "C:\ProgramData\chocolatey\lib-bad\winscp.install".
2024-02-20 13:41:37,475 13516 [DEBUG] - Sending message 'HandlePackageResultCompletedMessage' out if there are subscribers...
2024-02-20 13:41:37,478 13516 [ERROR] - The upgrade of winscp.install was NOT successful.
2024-02-20 13:41:37,481 13516 [ERROR] - Error while running 'C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyInstall.ps1'.
 See log for details.
2024-02-20 13:41:37,485 13516 [DEBUG] - Attempting to create directory "C:\ProgramData\chocolatey\lib-bad\winscp.install".
2024-02-20 13:41:37,486 13516 [DEBUG] - Moving C:\ProgramData\chocolatey\lib\winscp.install to C:\ProgramData\chocolatey\lib-bad\winscp.install\6.3.0
2024-02-20 13:41:37,488 13516 [DEBUG] - Moving 'C:\ProgramData\chocolatey\lib\winscp.install'
 to 'C:\ProgramData\chocolatey\lib-bad\winscp.install\6.3.0'
2024-02-20 13:41:39,495 13516 [DEBUG] - Attempting rollback
2024-02-20 13:41:39,497 13516 [DEBUG] - Moving C:\ProgramData\chocolatey\lib-bkp\winscp.install\6.1.2 to C:\ProgramData\chocolatey\lib\winscp.install
2024-02-20 13:41:39,498 13516 [DEBUG] - Moving 'C:\ProgramData\chocolatey\lib-bkp\winscp.install\6.1.2'
 to 'C:\ProgramData\chocolatey\lib\winscp.install'
2024-02-20 13:41:41,503 13516 [DEBUG] - Attempting to delete directory "C:\ProgramData\chocolatey\lib-bkp\winscp.install".
2024-02-20 13:41:41,520 13516 [WARN ] - 
Chocolatey upgraded 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
2024-02-20 13:41:41,522 13516 [INFO ] - 
2024-02-20 13:41:41,525 13516 [ERROR] - Failures
2024-02-20 13:41:41,527 13516 [ERROR] -  - winscp.install (exited -1) - Error while running 'C:\ProgramData\chocolatey\lib\winscp.install\tools\chocolateyInstall.ps1'.
 See log for details.
2024-02-20 13:41:41,529 13516 [DEBUG] - Sending message 'PostRunMessage' out if there are subscribers...
2024-02-20 13:41:41,532 13516 [DEBUG] - Exiting with -1
```


### Anything else?

_No response_

--- a/automatic/winscp.install/tools/chocolateyInstall.ps1
+++ b/automatic/winscp.install/tools/chocolateyInstall.ps1
@@ -5,8 +5,8 @@ $toolsPath = Split-Path $MyInvocation.MyCommand.Definition
 $packageArgs = @{
   packageName    = 'winscp'
   fileType       = 'exe'
-  file           = Get-Item $toolsPath\*.exe
-  file64         = Get-Item $toolsPath\*.exe
+  file           = "$toolsPath\WinSCP-$($env:ChocolateyPackageVersion)-Setup.exe"
+  file64         = "$toolsPath\WinSCP-$($env:ChocolateyPackageVersion)-Setup.exe"
   silentArgs     = '/VERYSILENT /SUPPRESSMSGBOXES /NORESTART /SP-'
 }
 Install-ChocolateyInstallPackage @packageArgs

--- a/automatic/winscp.install/tools/chocolateyInstall.ps1
+++ b/automatic/winscp.install/tools/chocolateyInstall.ps1
@@ -5,8 +5,8 @@ $toolsPath = Split-Path $MyInvocation.MyCommand.Definition
 $packageArgs = @{
   packageName    = 'winscp'
   fileType       = 'exe'
-  file           = "$toolsPath\WinSCP-$($env:ChocolateyPackageVersion)-Setup.exe"
-  file64         = "$toolsPath\WinSCP-$($env:ChocolateyPackageVersion)-Setup.exe"
+  file           = "$toolsPath\WinSCP-6.3.4-Setup.exe"
+  file64         = "$toolsPath\WinSCP-6.3.4-Setup.exe"
   silentArgs     = '/VERYSILENT /SUPPRESSMSGBOXES /NORESTART /SP-'
 }
 Install-ChocolateyInstallPackage @packageArgs

--- a/automatic/winscp.install/update.ps1
+++ b/automatic/winscp.install/update.ps1
@@ -4,11 +4,22 @@ $releases = 'https://winscp.net/eng/downloads.php'
 $re  = 'WinSCP.+\.exe/download$'
 
 function global:au_SearchReplace {
+    $FileVersion = if ($Latest.FileVersion -ne $Latest.Version) {
+        $Latest.FileVersion
+    } else {
+        '$($env:ChocolateyPackageVersion)'
+    }
+
    @{
         "$($Latest.PackageName).nuspec" = @{
             "(\<releaseNotes\>).*?(\</releaseNotes\>)" = "`${1}$($Latest.ReleaseNotes)`$2"
         }
-
+        "tools\chocolateyInstall.ps1"   = @{
+            # We use ChocolateyPackageVersion to minimise changes to the package code -
+            # If using a patchfix version, use the fileversion instead of the packageversion.
+            '(?i)(\s*file\s*=\s*["'']\$toolsPath\\WinSCP-)(\$\(\$env:ChocolateyPackageVersion\)|.+)(-Setup\.exe["''])' = "`${1}$($FileVersion)`${3}"
+            '(?i)(\s*file64\s*=\s*["'']\$toolsPath\\WinSCP-)(\$\(\$env:ChocolateyPackageVersion\)|.+)(-Setup\.exe["''])' = "`${1}$($FileVersion)`${3}"
+        }
         ".\legal\VERIFICATION.txt" = @{
           "(?i)(\s+x32:).*"            = "`${1} $($Latest.URL32)"
           "(?i)(checksum32:).*"        = "`${1} $($Latest.Checksum32)"
@@ -27,6 +38,7 @@ function global:au_GetLatest {
     $file_name = $url -split '/' | Select-Object -last 1 -Skip 1
     @{
         Version      = $version
+        FileVersion  = $version
         URL32        = "https://sourceforge.net/projects/winscp/files/WinSCP/$version/$file_name/download"
         FileName32   = $file_name
         ReleaseNotes = "https://winscp.net/download/WinSCP-${version}-ReadMe.txt"

--- a/automatic/winscp.install/update.ps1
+++ b/automatic/winscp.install/update.ps1
@@ -4,21 +4,13 @@ $releases = 'https://winscp.net/eng/downloads.php'
 $re  = 'WinSCP.+\.exe/download$'
 
 function global:au_SearchReplace {
-    $FileVersion = if ($Latest.FileVersion -ne $Latest.Version) {
-        $Latest.FileVersion
-    } else {
-        '$($env:ChocolateyPackageVersion)'
-    }
-
-   @{
+    @{
         "$($Latest.PackageName).nuspec" = @{
             "(\<releaseNotes\>).*?(\</releaseNotes\>)" = "`${1}$($Latest.ReleaseNotes)`$2"
         }
         "tools\chocolateyInstall.ps1"   = @{
-            # We use ChocolateyPackageVersion to minimise changes to the package code -
-            # If using a patchfix version, use the fileversion instead of the packageversion.
-            '(?i)(\s*file\s*=\s*["'']\$toolsPath\\WinSCP-)(\$\(\$env:ChocolateyPackageVersion\)|.+)(-Setup\.exe["''])' = "`${1}$($FileVersion)`${3}"
-            '(?i)(\s*file64\s*=\s*["'']\$toolsPath\\WinSCP-)(\$\(\$env:ChocolateyPackageVersion\)|.+)(-Setup\.exe["''])' = "`${1}$($FileVersion)`${3}"
+            "(?i)(^\s*file\s*=\s*`"[$]toolsPath\\).*"   = "`${1}$($Latest.FileName32)`""
+            "(?i)(^\s*file64\s*=\s*`"[$]toolsPath\\).*" = "`${1}$($Latest.FileName32)`""
         }
         ".\legal\VERIFICATION.txt" = @{
           "(?i)(\s+x32:).*"            = "`${1} $($Latest.URL32)"
@@ -38,7 +30,6 @@ function global:au_GetLatest {
     $file_name = $url -split '/' | Select-Object -last 1 -Skip 1
     @{
         Version      = $version
-        FileVersion  = $version
         URL32        = "https://sourceforge.net/projects/winscp/files/WinSCP/$version/$file_name/download"
         FileName32   = $file_name
         ReleaseNotes = "https://winscp.net/download/WinSCP-${version}-ReadMe.txt"

